[16.0][ADD] website_sale_loyalty_suggestion_wizard_multi_gift: New Module

pilarvargas-tecnativa opened the pull request:
cc @tecnativa TT48349

@chienandalu @carlos-lopez-tecnativa  please review

pilarvargas-tecnativa committed:
[ADD] website_sale_loyalty_suggestion_wizard_multi_gift: New Module

TT48349

carlos-lopez-tecnativa commented on website_sale_loyalty_suggestion_wizard_multi_gift/templates/wizard_templates.xml:
```diff
@@ -0,0 +1,47 @@
+                            t-as="product"
+                        >
+                            <t
+                                t-call="website_sale_loyalty_suggestion_wizard.promotion_wizard_reward_product"
+                            >
+                                <t
+                                    t-set="reward_qty"
+                                    t-value="multi_gift_line.reward_product_quantity"
+                                />
+                                <t t-set="product" t-value="product" />
+                                <t t-set="id" t-value="multi_gift_line.id" />
+                                <t
+                                    t-set="optional_product"
+                                    t-value="len(multi_gift_line.reward_product_ids) > 1"
+                                />
+                                <t
+                                    t-set="option_checked"
+                                    t-value="product == multi_gift_line.reward_product_ids[:1]"
+                                />
```
Maybe in another module or in this one, but I think it could be useful to add the product name including attributes to differentiate between variants. What do you think?

![image](https://github.com/user-attachments/assets/add446b7-c0da-496e-87cc-c5879f574ea1)

pilarvargas-tecnativa replied:
Done!

![image](https://github.com/user-attachments/assets/b3e57524-8667-4570-b5bb-3ab76799a0ef)

carlos-lopez-tecnativa reviewed (changes requested):
Functionally tested, just a minor change to make it clearer which product is selected when using variants.

pilarvargas-tecnativa committed:
[IMP] website_sale_loyalty_suggestion_wizard: Show full product name

When setting up a multi-gift promotion or even in a simple promotion,
showing the product name without differentiating that it may be a variant,
the customer does not know exactly what gift they are receiving.
To avoid this confusion and also when choosing alternatives in a
multi-gift promotion, the full name is shown with its attributes but
omitting the reference.

chienandalu commented on website_sale_loyalty_suggestion_wizard_multi_gift/controllers/promotion_wizard.py:
```diff
@@ -0,0 +1,16 @@
+        response = super()._process_reward_line_options(wizard_id, reward_line_options)
+        if wizard_id.multi_gift_reward:
+            selected_product_ids = reward_line_options.get("selected_product_ids")
+            for i in range(len(selected_product_ids)):
+                wizard_id.loyalty_gift_line_ids[i].selected_gift_id = int(
+                    selected_product_ids[i]
+                )
```
This loop feels a little bit weird. Without knowing the precise internals I'd say you could better use `zip` for the task.

pilarvargas-tecnativa replied:
done, thanks!

carlos-lopez-tecnativa reviewed (approved):
LGTM

pedrobaeza commented:
/ocabot merge nobump
